### PR TITLE
[Release] Add FAQ explaining release terms

### LINKED ISSUE #98009: Explain what cherry-pick and branch cut terms mean in RELEASE.md

### PR DESCRIPTION
**Pull request by atalman**
[Release] Add FAQ explaining release terms
This is AI from release 2.0.0 retrospective
Fixes: https://github.com/pytorch/pytorch/issues/98009

### CODE REVIEW

File: RELEASE.md [hunk 1]
```diff
@@ -55,6 +56,14 @@ Releasing a new version of PyTorch generally entails 3 major steps:
 2. Drafting RCs (Release Candidates), and merging cherry picks
 3. Promoting RCs to stable and performing release day tasks
 
+### Frequently Asked Questions
+
+* Q: What is release branch cut  ?
+  * A: With almost all of the feature PRs merged into main, the release engineer cuts a release branch such that we can build a set of release candidates (RC). The purpose of this stage is stabilization of the overall PyTorch code base with massive changes introduced by new features. The activities will focus on testing, gathering feedback and pipe-cleaning the release.
```
Comment (malfet) [inline RELEASE.md:62]:
```suggestion
  * A: When bulk of the tracked feature merged into main branch, the release engineer starts the release process by creates a branch to prepare for a stable release. This allows PyTorch development flow to continue uninterrupted, while release engineering team on stabilizing the features and testing them in a series of  release candidates (RC). The activities in release branch will focus on testing, polishing new features and fixing regression. In general, new features are not supposed to be added to the branch after it was created.
```

File: RELEASE.md [hunk 2]
```diff
@@ -55,6 +56,14 @@ Releasing a new version of PyTorch generally entails 3 major steps:
...
+
+* Q: What is cherry-pick ?
+  * A: A cherry pick is a process to land a PR in the release branch after branch cut. These are typically limited to ensure that the team has sufficient time to complete a thorough round of testing on a stable code base. Please see issue [tracker](https://github.com/pytorch/pytorch/issues/94937) for cherry-picking acceptance criteria and details.
```
[the remark this answers is not in the document]
Comment (atalman) [reply RELEASE.md]: Reworded without mentioning landing

File: RELEASE.md [hunk 3]
```diff
@@ -55,6 +56,14 @@ Releasing a new version of PyTorch generally entails 3 major steps:
 2. Drafting RCs (Release Candidates), and merging cherry picks
 3. Promoting RCs to stable and performing release day tasks
 
+### Frequently Asked Questions
+
+* Q: What is release branch cut  ?
+  * A: When bulk of the tracked features merged into the main branch, the release engineer starts the release process by creating the release branch. This allows PyTorch development flow to continue uninterrupted, while release engineering team focuses on stabilizing the features and testing them in a series of  release candidates (RC). The activities in release branch includes regression and performance testing, polishing new features and fixing release-specific bugs. In general, new features are not supposed to be added to the release branch after it was created.
```
Comment (malfet) [inline RELEASE.md:62]:
```suggestion
  * A: When bulk of the tracked features merged into the main branch, the release engineer starts the release process by creating the release branch. This allows PyTorch development flow to continue uninterrupted, while release engineering team focuses on stabilizing the features and testing them in a series of release candidates (RC). The activities in release branch includes regression and performance testing, polishing new features and fixing release-specific bugs. In general, new features are not supposed to be added to the release branch after it was created.
```

File: RELEASE.md [hunk 4]
```diff
@@ -55,6 +56,14 @@ Releasing a new version of PyTorch generally entails 3 major steps:
...
+
+* Q: What is cherry-pick ?
+  * A: A cherry pick is a process of propagating commits from the main into the  release branch. Those commits are typically small fixes or  documentation updates to ensure that the team has sufficient time to complete a thorough round of testing on a stable code base. To nominate fix for cherry-pick, separate pull request must be created against respective release branch and PR itself, along with its trunk counterpart must be mentioned in Release Tracker issue using the template outlined in the issue description. For example see [Release tracker for 2.0.0 release](https://github.com/pytorch/pytorch/issues/94937).
```
Comment (malfet) [inline RELEASE.md:65]:
```suggestion
  * A: A cherry pick is a process of propagating commits from the main into the release branch. Those commits are typically small fixes or documentation updates to ensure that the team has sufficient time to complete a thorough round of testing on a stable code base. To nominate fix for cherry-pick, separate pull request must be created against respective release branch and then mentioned in Release Tracker issue following the template from the issue description. That is  comment nominating cherry-pick for inclusion in release,  should include committed PR against main branch, cherry-pick PR and acceptance criteria. For example see [Release tracker for 2.0.0 release](https://github.com/pytorch/pytorch/issues/94937).
```

**Review by malfet — APPROVED** [review]

File: RELEASE.md [hunk 5]
```diff
@@ -55,6 +56,14 @@ Releasing a new version of PyTorch generally entails 3 major steps:
 2. Drafting RCs (Release Candidates), and merging cherry picks
 3. Promoting RCs to stable and performing release day tasks
 
+### Frequently Asked Questions
+
+* Q: What is release branch cut  ?
+  * A: When bulk of the tracked features merged into the main branch, the release engineer starts the release process by creating the release branch. This allows PyTorch development flow to continue uninterrupted, while release engineering team focuses on stabilizing the features and testing them in a series of release candidates (RC). The activities in release branch includes regression and performance testing, polishing new features and fixing release-specific bugs. In general, new features are not supposed to be added to the release branch after it was created.
```
Comment (seemethere) [inline RELEASE.md:62]:
```suggestion
  * A: When bulk of the tracked features merged into the main branch, the primary release engineer starts the release process of cutting the release branch by creating a new git branch based off of the current `main` development branch of PyTorch. This allows PyTorch development flow on `main` to continue uninterrupted, while the release engineering team focuses on stabilizing the release branch in order to release a series of release candidates (RC). The activities in the release branch include both regression and performance testing as well as polishing new features and fixing release-specific bugs. In general, new features *are not* added to the release branch after it was created.
```

**Review by seemethere — COMMENTED** [review]
Some general improvements

**Pull request closed without merging**